### DISCUSSION TITLE
Drawing polygons/shapes using negative values

### DISCUSSION THREAD

**Discussion by JonnyBooker** [SixLabors/ImageSharp.Drawing#188] [Q&A] [answered]
I am currently in the process of migrating from System.Drawing to ImageSharp and I'm having some unexpected side effects I'm not sure how to overcome.

The problem I am faced with is when doing "Drawing", if the circle previously went outside the bounds of the Bitmap, this was previously fine and it would draw the parts of the circle it could, simply clipping the drawn circle. With ImageSharp, this throws an out of bounds exception.

This issue can be replicated using this code:
```csharp
using var image = new Image<Rgba32>(64, 64);
var ellipse = new EllipsePolygon(32, 32, 70, 70);

var points = ellipse.Points.ToArray();
image.Mutate(x => x.Draw(SixLabors.ImageSharp.Color.Green, 1f, ellipse));

using var memoryStream = new MemoryStream();
image.Save(memoryStream, new JpegEncoder());
	
Util.Image(memoryStream.ToArray()).Dump();
```

Which gives:
> ArgumentOutOfRangeException: Specified argument was out of the range
> of valid values. (Parameter 'edgeIdx')

Is there a way that this would be possible at all? Such as configuration that would allow me to draw all the shape that I can? This is the desired effect that I could achieve using System.Drawing:

[![enter image description here][1]][1]

  [1]: https://i.stack.imgur.com/GVI56.png
  [2]: https://i.stack.imgur.com/xTK3n.png

**Comment by JimBobSquarePants** [comment, marked as answer]
This looks like it could be related to #170 could you please try the MyGet build 1.0.0-beta13.17 to see if we’ve already fixed it?

**Reply by JimBobSquarePants** [reply]
@JonnyBooker How did you get on?

**Reply by JonnyBooker** [reply]
@JimBobSquarePants Very sorry! Not been well and not had chance to fully investigate. I did manage to do some investigation after starting this discussion and pulled down the latest source to try it against source so could debug the issue myself and did see that it worked.

I integrated the MyGet package you suggested does fix it, thank you! Will use that version for the moment in which case!

**Reply by JimBobSquarePants** [reply]
Excellent. Thanks for confirming 👍 We'll be looking to get a release out on NuGet proper ASAP.
